tutu-frontend/src/app/hakemus/[oid]/perustelu/yleiset/perustelut/components/Lahde.tsx

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useState } from 'react';
+
+import { Stack, useTheme } from '@mui/material';
+import { OphCheckbox, OphTypography } from '@opetushallitus/oph-design-system';
+import { useTranslations } from '@/src/lib/localization/hooks/useTranslations';
+import { Perustelu } from '@/src/lib/types/perustelu';
+
+interface Props {
+  perustelu: Perustelu | undefined;
+  updatePerustelu: (perustelu: Perustelu) => void;
+}
+
+export const Lahde = ({ perustelu, updatePerustelu }: Props) => {
+  const { t } = useTranslations();
+  const theme = useTheme();
+
+  const [isLahtomaanKansallinenLahde, setIsLahtomaanKansallinenLahde] =
+    useState<boolean>(false);
+
+  const [isLahtomaanVirallinenVastaus, setIsLahtomaanVirallinenVastaus] =
+    useState<boolean>(false);
+
+  const [
+    isKansainvalinenHakuteosTaiVerkkosivusto,
+    setIsKansainvalinenHakuteosTaiVerkkosivusto,
+  ] = useState<boolean>(false);
+
+  const updateIsLahtomaanKansallinenLahde = (val: boolean) => {
+    setIsLahtomaanKansallinenLahde(val);
+    const _perustelu = perustelu ?? ({} as Perustelu);
+    updatePerustelu({
+      ..._perustelu,
+      lahdeLahtomaanKansallinenLahde: val,
+    });
+  };
```

Comment:
```suggestion
  const updateValue = (field: string, val: boolean) => {
    setIsLahtomaanKansallinenLahde(val);
    const _perustelu = perustelu ?? ({} as Perustelu);
    updatePerustelu({
      ..._perustelu,
     [field]: val,
    });
  };
```
periaatteessa vois kirjottaa näihin vaan yhen funktion ja kutsua updatessa `updateValue('lahtomaanKansallinenLahde', event.target.value)` tms